src/lib/input/autosize.ts

```diff
@@ -136,10 +136,13 @@ export class MdTextareaAutosize implements AfterViewInit {
     }
 
     // Reset the textarea height to auto in order to shrink back to its default size.
+    // Also temporarily force overflow:hidden, so scroll bars do not interfere with calculations.
     textarea.style.height = 'auto';
+    textarea.style.overflow = 'hidden';
```

Comment:
could find some text that previously wasn't calculated right and check that its height is correct now I suppose